pkg/extensions/search/common/oci_layout.go

```diff
@@ -238,15 +253,120 @@ func (olu OciLayoutUtils) checkCosignSignature(name, digest string) bool {
 // checks if manifest is signed or not
 // checks for notary or cosign signature
 // if cosign signature found it does not looks for notary signature.
-func (olu OciLayoutUtils) checkManifestSignature(name, digest, mediaType string) bool {
+func (olu BaseOciLayoutUtils) checkManifestSignature(name, digest, mediaType string) bool {
 	if !olu.checkCosignSignature(name, digest) {
 		return olu.checkNotarySignature(name, digest, mediaType)
 	}
 
 	return true
 }
 
-func (olu OciLayoutUtils) GetExpandedRepoInfo(name string) (RepoInfo, error) {
+func (olu BaseOciLayoutUtils) GetImageLastUpdated(repo string, manifestDigest godigest.Digest) time.Time {
```

Comment:
I think it is good idea to take imageInfo as an argument or define a new type because almost for all methods below we are first obtaining ImageInfo. 
```suggestion
func (olu OciLayoutUtils) GetImageLastUpdated(repo string, imageInfo ispec.Image) time.Time {
```
or 

``` 
type imageInfo ispec.ImageInfo 

func (imgInfo imageInfo) GetTimeStamp() time.Time {
        var timeStamp time.Time

	if len(imgInfo.History) != 0 {
		timeStamp = *imgInfo.History[0].Created
	} else {
		timeStamp = time.Time{}
	}

	return timeStamp
} 
```

Similar for all below methods as required.